### PR TITLE
Only hash

### PR DESCRIPTION
**Pull request by bmordan**
```sh
-n,         --only-hash     bool   - Only chunk and hash - do not write to disk.
```

Here are the changes to make this so. There are tests for **builder** and **Importer**.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,6 +79,7 @@
     "pull-stream": "^3.6.0",
     "pull-traverse": "^1.0.3",
     "pull-write": "^1.1.4",
+    "readable-stream": "^2.3.3",
```
Comment (alanshaw) [inline package.json]: I don't believe these dependencies are needed for your work

File: src/importer/tree-builder.js [hunk 2]
```diff
@@ -14,14 +14,14 @@ module.exports = createTreeBuilder
 
 const defaultOptions = {
   wrap: false,
-  shardSplitThreshold: 1000
+  shardSplitThreshold: 1000,
+  onlyHash: true
```
Comment (alanshaw) [inline src/importer/tree-builder.js]: The default should be false

File: test/node.js [hunk 3]
```diff
@@ -53,4 +53,5 @@ describe('IPFS UnixFS Engine', () => {
   require('./test-nested-dir-import-export')(repo)
   require('./test-dirbuilder-sharding')(repo)
   require('./test-dag-api')
+  require('./test-only-hash')(repo)
```
Comment (alanshaw) [inline test/node.js:56]: Please add to `test/browser.js` also

File: test/test-importer.js [hunk 4]
```diff
@@ -417,6 +418,37 @@ module.exports = (repo) => {
           }
         }
       })
+
+      it.only('will not write to disk if passed --only-hash option', (done) => {
```
Comment (alanshaw) [inline test/test-importer.js]: Please remove `.only`

Comment (alanshaw) [reply test/test-importer.js]: The option name at this level is `onlyHash` not `--only-hash`

File: test/test-only-hash.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+module.exports = (repo) => {
+  describe('builder', () => {
+    let ipldResolver
+
+    before(() => {
+      const bs = new BlockService(repo)
+      ipldResolver = new IPLDResolver(bs)
+    })
+
+    it('will only chunk and hash if passed an --only-hash option', (done) => {
```
Comment (alanshaw) [inline test/test-only-hash.js]: The option name at this level is `onlyHash` not `--only-hash`

**Review by alanshaw — CHANGES_REQUESTED** [review]
Please add `onlyHash` option to the README

File: test/node.js (hunk 3 above)
Comment (bmordan) [reply test/node.js:56]: Have done this. See above.

**Review by alanshaw — APPROVED** [review]

**Comment by alanshaw** [comment]
Opened against ipfs/js-ipfs-unixfs-engine

**Pull request closed without merging**